ecs-agent/acs/session/attach_resource_responder_test.go

```diff
@@ -0,0 +1,147 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/testconst"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/api/resource"
+)
+
+const (
+	testAttachmentArn = "arn:aws:ecs:us-west-2:123456789012:ephemeral-storage/a1b2c3d4-5678-90ab-cdef-11111EXAMPLE"
+	testClusterArn    = "arn:aws:ecs:us-west-2:123456789012:cluster/a1b2c3d4-5678-90ab-cdef-11111EXAMPLE"
+)
+
+var (
+	testAttachmentProperties = []*ecsacs.AttachmentProperty{
+		{
+			Name:  aws.String(resource.FargateResourceIdName),
+			Value: aws.String("name1"),
+		},
+		{
+			Name:  aws.String(resource.VolumeIdName),
+			Value: aws.String("id1"),
+		},
+		{
+			Name:  aws.String(resource.VolumeSizeInGiBName),
+			Value: aws.String("size1"),
+		},
+		{
+			Name:  aws.String(resource.RequestedSizeName),
+			Value: aws.String("size2"),
+		},
+		{
+			Name:  aws.String(resource.ResourceTypeName),
+			Value: aws.String(resource.EphemeralStorage),
+		},
+		{
+			Name:  aws.String(resource.DeviceName),
+			Value: aws.String("device1"),
+		},
+	}
+	testAttachment = &ecsacs.Attachment{
+		AttachmentArn:        aws.String(testAttachmentArn),
+		AttachmentProperties: testAttachmentProperties,
+	}
+	testConfirmAttachmentMessage = &ecsacs.ConfirmAttachmentMessage{
+		Attachment:           testAttachment,
+		MessageId:            aws.String(testconst.MessageID),
+		ClusterArn:           aws.String(testClusterArn),
+		ContainerInstanceArn: aws.String(testconst.ContainerInstanceARN),
+		TaskArn:              aws.String(testconst.TaskARN),
+		WaitTimeoutMs:        aws.Int64(testconst.WaitTimeoutMillis),
+	}
+)
+
```

Comment:
Q: Should we be testing the creation of a new a new AttachResourceResponder as well as the `handleAttachMessage()` functionality somewhere?